src/java/org/apache/cassandra/service/accord/serializers/TableMetadatas.java

```diff
@@ -0,0 +1,283 @@
+    public abstract long serializedSelfSize();
+
+    public static TableMetadatas none()
+    {
+        return Multi.NONE;
+    }
+
+    public static TableMetadatas of(TableMetadata metadata)
+    {
+        return new One(metadata);
+    }
+
+    public static TableMetadatas of(TableMetadata ... metadatas)
+    {
+        if (metadatas.length == 0)
+            return none();
+        if (metadatas.length == 1)
+            return new One(metadatas[0]);
+        return new Multi(metadatas);
+    }
+
+    static class One extends TableMetadatas
+    {
+        final TableMetadata table;
+
+        One(TableMetadata table)
+        {
+            this.table = table;
+        }
+
+        @Override
+        public TableMetadata get(int index)
+        {
+            Invariants.require(index == 0);
+            return table;
+        }
+
+        @Override
+        public int size()
+        {
+            return 1;
+        }
+
+        @Override
+        public int indexOf(TableMetadata find)
+        {
+            if (find.id.equals(table.id))
+                return 0;
+            return -1;
+        }
+
+        @Override
+        public void serialize(TableMetadata table, DataOutputPlus out) throws IOException
+        {
+        }
+
+        @Override
+        public void serializeSelf(DataOutputPlus out) throws IOException
+        {
+            out.writeUnsignedVInt32(1);
+            table.id.serializeCompactComparable(out);
+        }
+
+        @Override
+        public TableMetadata deserialize(DataInputPlus in) throws IOException
+        {
+            return table;
+        }
+
+        @Override
+        public long serializedSize(TableMetadata table)
+        {
+            return 0;
+        }
+
+        @Override
+        public long serializedSelfSize()
+        {
+            return TypeSizes.sizeofUnsignedVInt(1) + table.id.serializedCompactComparableSize();
+        }
+
+        @Override
+        public int indexOf(TableId tableId)
+        {
+            if (tableId.equals(table.id))
+                return 0;
+            return -1;
+        }
+
+        @Override
+        public TableMetadata get(TableId tableId)
+        {
+            if (tableId.equals(table.id))
+                return table;
+            return null;
+        }
+    }
+
+    static class Multi extends TableMetadatas
+    {
+        static final TableMetadatas NONE = new Multi();
+
+        final TableMetadata[] tables;
+
+        Multi(TableMetadata ... tables)
+        {
+            this.tables = tables;
+        }
+
+        @Override
+        public TableMetadata get(int index)
+        {
+            return tables[index];
+        }
+
+        @Override
+        public int size()
+        {
+            return tables.length;
+        }
+
+        @Override
+        public int indexOf(TableMetadata find)
+        {
+            return Arrays.binarySearch(tables, find, comparingId);
```

Comment:
this can return `< -1`, where as `One` can only return `-1`; *should* they have consistent behavior?

Reply:
We only treat less than zero as not present, but I don’t mind fulfilling binary search semantics if you prefer for One; or could just document less than zero only means not present. Don’t mind

Reply:
i am fine with either.  I know many `indexOf` methods document `-1` is the false case, where as we are `< 0`.  I am personally fine with documentation or binary search semantics.